src/state_machine/sm.cpp

```diff
@@ -175,6 +176,8 @@ BULLSEYE_EXCLUDE_BLOCK_START
 					return ERROR;
 				}
 
+				SM_ASSERT_POINTER(m_p_sm_table[st].event_info);
```

Comment:
`SM_ASSERT_POINTER(m_p_sm_table[st].event_info)` is called immediately after the calloc() call (Line #119). 
But you think it is still needed here?